### ISSUE TITLE
How to seperate `Tracker` into multiple modules?

### ISSUE THREAD

**Issue by dev-matan-tsuberi** [daostack/daostack-node#3]
`Tracker` currently has a lot of responsibilities:
1. fetch data from blockchain.
2. write updates to DB
3. expose a REST API
4. publish updates to web-sockets
5. In the future we may have other stuff like sending emails, tweeting etc.

difficulties in separating the different responsibilities:
1. `fetch` needs to know the last synced block and updating the last synced block should happen within the same transaction as all the `write`s that happen as a result of it.
2. The REST API and `write` both need to know about the SQL schema and run queries.
3. publishing updates needs to happen after the transaction has been committed to the DB.

**Comment by tibetsprague** [comment]
When you are defining the models you add a postgresql section that looks like:

 "postgresql": {
        "columnName": "id",
        "dataType": "integer",
        "dataLength": null,
        "dataPrecision": 32,
        "dataScale": 0,
        "nullable": "NO"
      }

see https://loopback.io/doc/en/lb3/Connecting-to-PostgreSQL.html

**Comment by tsuberim** [comment]
Yes, I saw that, but the question is what's the loopback type for that? because a JS `number` is going to cause a loss of precision.
